monasca_rsyslog/client.py

```diff
@@ -60,11 +72,27 @@ def __init__(self):
             auth=auth_plugin,
             user_agent='rsyslog-monasca'
         )
-        self._url = cfg.CONF.api.url
+        self._url = self._get_monasca_log_api_url()
         self._verbosity = cfg.CONF.api.verbosity
         self._min_poll_delay = cfg.CONF.api.min_poll_delay
         self._max_batch_size = cfg.CONF.api.max_batch_size
 
+    def _get_monasca_log_api_url(self, version='/v3.0'):
+        """ Retrieves monasca log api endpoint. """
+
+        endpoint = cfg.CONF.api.url
+        if not endpoint:
+            args = {
+                'service_type': cfg.CONF.auth.service_type,
+                'interface': cfg.CONF.auth.endpoint_type,
+                'region_name': cfg.CONF.auth.region_name  # that one has no default
+            }
+            catalog = self._sess.auth.get_auth_ref(self._sess).service_catalog
+            endpoint = catalog.url_for(**args)
+        if not endpoint.endswith(version):
```

Comment:
I think really we should be able to make this work without the version, but we can consider this outside of the scope of this change.

Reply:
Hmm what do you mean? Without the version, the posting fails.

Reply:
The Log API can return a list of endpoints, and indicate which one is the most current, for example:

```
# curl 192.168.1.1:5607
{
  "elements": [
    {
      "id": "v3.0",
      "links": [
        {
          "href": "http:192.168.1.1:5607/v3.0/logs",
          "rel": "logs"
        }
      ],
      "status": "CURRENT",
      "updated": "2016-03-01T00:00:00Z"
    },
    {
      "id": "v2.0",
      "links": [
        {
          "href": "http:192.168.1.1:5607/v2.0/log/single",
          "rel": "logs"
        }
      ],
      "status": "DEPRECATED",
      "updated": "2015-09-01T00:00:00Z"
    }
  ],
  "links": [
    {
      "href": "http://192.168.1.1:5607/",
      "rel": "self"
    },
    {
      "href": "http:192.168.1.1:5607/version",
      "rel": "version"
    },
    {
      "href": "http:192.168.1.1:5607/healthcheck",
      "rel": "healthcheck"
    }
  ]
}
```